testsuite/MDAnalysisTests/formats/test_libdcd.py

```diff
@@ -242,6 +243,22 @@ def test_write_header(tmpdir):
         assert header['nsavc'] == 10
         assert np.allclose(header['delta'], .02)
 
+    # we also check the bytes written directly.
+    with open(testfile, 'rb') as fh:
+        header_bytes = fh.read()
+    # check for magic number
+    assert struct.unpack('i', header_bytes[:4])[0] == 84
```

Comment:
It's been a while since I had to reverse engineer Fortran binary, but this looks right-
 the magic numbers are the length in bytes of the arrays

Reply:
Yes, the magic number is the length of the data block. It looks like all data blocks in the DCD file are marked with numbers of the corresponding length at the beginning and end. As I can see the header of a DCD file has 3-4 blocks. 

1. *84 bytes* time information like dt, nsavc, ...
1. *x * 80+4 bytes* remarks
1. *4 bytes* natoms
1. *xx bytes* fixed atoms (may not be written)